src/game/logic/object/update/physicsupdate.cpp

```diff
@@ -13,17 +13,452 @@
  *            LICENSE
  */
 #include "physicsupdate.h"
+#include "audiomanager.h"
+#include "damage.h"
 #ifdef GAME_DLL
 #include "hooker.h"
 #endif
 #include "gamelogic.h"
+#include "globaldata.h"
 #include "opencontain.h"
+#include "terrainlogic.h"
+
+float Height_To_Speed(float height)
+{
+    return GameMath::Sqrt(GameMath::Fabs(2.0f * g_theWriteableGlobalData->m_gravity * height));
+}
+
+void Parse_Height_To_Speed(INI *ini, void *, void *store, const void *)
+{
+    *static_cast<float *>(store) = Height_To_Speed(ini->Scan_Real(ini->Get_Next_Token()));
+}
 
 void Parse_Friction_Per_Sec(INI *ini, void *, void *store, const void *)
 {
     *static_cast<float *>(store) = ini->Scan_Real(ini->Get_Next_Token()) * 1.0f / 30.0f;
 }
 
+PhysicsBehaviorModuleData::PhysicsBehaviorModuleData() :
+    m_mass(1.0f),
+    m_shockResitance(0),
+    m_shockMaxYaw(0.05f),
+    m_shockMaxPitch(0.025f),
+    m_shockMaxRoll(0.025f),
+    m_forwardFriction(0.15f),
+    m_lateralFriction(0.15f),
+    m_zFriction(0.8f),
+    m_aerodynamicCoeff(0.0f),
+    m_centerOfMassOffset(0),
+    m_allowBouncing(false),
+    m_allowCollideForce(true),
+    m_killWhenRestingOnGround(false),
+    m_minFallHeightForDamage(Height_To_Speed(40.0f)),
+    m_fallHeightDamageFactor(1.0f),
+    m_pitchRollYawFactor(2.0f),
+    m_vehicleCrashesIntoBuildingWeapon(g_theWeaponStore->Find_Weapon_Template("VehicleCrashesIntoBuildingWeapon")),
+    m_vehicleCrashesIntoNonBuildingWeapon(g_theWeaponStore->Find_Weapon_Template("VehicleCrashesIntoNonBuildingWeapon"))
+{
+}
+
+void PhysicsBehaviorModuleData::Build_Field_Parse(MultiIniFieldParse &p)
+{
+    static const FieldParse dataFieldParse[] = {
```

Comment:
look at how others like these are formatted and follow that formatting

Reply:
fixed.